src/main/java/com/google/sps/data/Referendum.java

```diff
@@ -49,9 +49,24 @@ public static Builder builder() {
 
   // creates a new Referendum object by extracting the properties from "obj"
   public static Referendum fromJSONObject(JSONObject obj) throws JSONException {
+    String referendumDescription;
+    String referendumTitle;
+
+    try {
+      referendumTitle = obj.getString(TITLE_JSON_KEYWORD);
+    } catch (JSONException e) {
+      throw new JSONException("Referendum title not found in JSON response.");
+    }
+
+    try {
+      referendumDescription = obj.getString(DESCRIPTION_JSON_KEYWORD);
+    } catch (JSONException e) {
+      referendumDescription = "";
```

Comment:
please add logging.  Why are we throwing an exception above but not here?

Reply:
The rationale is that it's possible for a referendum to be missing a referendumSubtitle field and in that case its sufficient to just catch the exception and fill in the description field with an emptyString. However, a missing referendumTitle probably indicates some sort of malformed response from the API or some undesired manipulation of the obj JSONObject which would warrant throwing an exception.  Will add logging.